docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+a simple coffee-shop service, generating TypeScript code for both the client and server, and implementing both a
+front-end and back-end for the service. We won't assume that you're an expert in Smithy, but it may helpful to work
+through the :doc:`../quickstart` before beginning this tutorial.
+
+Let's get started.
+
+Setting Up the Project
+======================
+This application will consist of a 4 major components:
+
+1. The model
+2. The server
+3. The client
+4. The web application
+
+--------------
+Pre-requisites
+--------------
+To follow this tutorial, you'll need to install a few tools.
+
+* :doc:`Smithy CLI <../guides/smithy-cli/cli_installation>`
+* `Node.js (>= 16) <https://nodejs.org/en/download>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_
+
+------
+Set up
+------
+Once you have these installed, the CLI has a useful command to easily bootstrap projects from
+`the repository of example smithy projects <https://github.com/smithy-lang/smithy-examples>`_. Execute the following
+command to set up the initial project:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    smithy init -t full-stack-application
+
+You will now have the project in the ``full-stack-application`` directory. In this project, you will find a ``README``
+that contains important information about this project, like how it is laid out and how to build and run it. In this
+tutorial, we will show exactly which commands to run and when.
+
+.. TODO: Provide the skeleton template or a git patch?
+
+Modeling Our Service
+====================
+With the basic framework for the project established, let's walk through how to model our coffee service.
+
+As discussed above, the coffee service should:
+
+* provide a menu of coffees
+* provide the ability to order a coffee
+* provide the ability to check the status of an order
+
+------------------
+Adding a Service
+------------------
+The service shape is the entry-point of our API, and is where we define the operations that our service exposes to a
+consumer. With that information in mind, let's define the initial service shape without any operations:
+
+.. code-block:: smithy
+    :caption: ``main.smithy``
+
+    $version: "2.0"
+
+    namespace com.example
+
+    use aws.protocols#restJson1
+
+    /// Allows users to retrieve a menu, create a coffee order, and
+    /// and to view the status of their orders
+    @title("Coffee Shop Service")
+    @restJson1
+    service CoffeeShop {
+        version: "2024-04-04"
+    }
+
+We apply the ``@restJson1`` protocol trait to the service to indicate that this service supports the
+:doc:`../aws/protocols/aws-restjson1-protocol` created and used extensively by AWS.
+Put simply, protocols define the rules and conventions for how data is serialized and deserialized when communicating
+between client and server. Protocols are complex and need their own topic, so we will not discuss them at length in
```

Comment:
```
between client and server. Protocols are addressed [HERE]()?, so we will not discuss them at length in
```
Not sure I would include this line about protocols being "complex"

Reply:
removing